### PR TITLE
DD-490: Also substitute depositor-account in deposit.properties

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes DD-490: Also substitute depositor-account in deposit.properties

When applied it will also
--------------------
* [x] show the change with a unit test
* fixed #46 `amd.xml` not in data/easy-migration
* fixed mismatches between jumpoff and ThemathischeCollecties.csv

Where should the reviewer @DANS-KNAW/easy @DANS-KNAW/dataversedans  start?
------------------------------------------------

How should this be manually tested?
-----------------------------------

* on host, the  AppSpec unit tests one by one, then

      cp -r easy-convert-bag-to-deposit/target/test/AppSpec/ingest-dir/*/* easy-dtap/shared/

  on deasy (fill in one of the copied bags)

      easy-validate-dans-bag /vagrant/shared/*/

  The bags are valid.

* [X] round trip with a manually deposited dataset ('easy-dataset:18' in the script below) and compiled/deployed modules  on deasy: `fedora-to-bag`, `bag-to-deposit`, `validate-dans-bag`

      sudo mv /etc/opt/dans.knaw.nl/easy-convert-bag-to-deposit/ThemathischeCollecties.csv ThemathischeCollecties-bu.csv
      sudo touch /etc/opt/dans.knaw.nl/easy-convert-bag-to-deposit/ThemathischeCollecties.csv
      sudo chmod 777 /etc/opt/dans.knaw.nl/easy-convert-bag-to-deposit/
      rm -rf easy-*.log /vagrant/shared/btf-out/* /vagrant/shared/easy-fedora-to-bag-staging/*
      easy-fedora-to-bag -d 'easy-dataset:18' -o /vagrant/shared/btf-out -f SIP simple # ID of a new manual deposit
      easy-convert-bag-to-deposit -d /vagrant/shared/btf-out/ -t DOI -s FEDORA
      easy-validate-dans-bag --aip /vagrant/shared/btf-out/UUID/UUID # fill in UUIDs with autocompletion

Related pull requests on github
-------------------------------

replaces #49

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala [hunk 1]
```diff
@@ -80,7 +81,13 @@ class EasyConvertBagToDepositApp(configuration: Configuration) extends DebugEnha
   private def addProps(depositPropertiesFactory: DepositPropertiesFactory, maybeOutputDir: Option[File])
                       (bagParentDir: File): Try[Boolean] = {
     logger.debug(s"creating application.properties for $bagParentDir")
-    val changedMetadata = Seq("bag-info.txt", "metadata/amd.xml", "metadata/dataset.xml", "metadata/provenance.xml").map(Paths.get(_))
+    val changedMetadata = Seq(
+      "bag-info.txt",
+      "metadata/amd.xml", "metadata/emd.xml", // in practice one of both
```
Comment (lindareijnhoudt) [inline src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala:86]:
```suggestion
      "metadata/amd.xml", 
      "metadata/emd.xml", 
```
is this is leftover of the amd/emd confusion?

File: src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala [hunk 2]
```diff
@@ -91,29 +98,32 @@ class EasyConvertBagToDepositApp(configuration: Configuration) extends DebugEnha
...
       newDcmi = (ddmOut \ "dcmiMetadata").headOption.getOrElse(<dcmiMetadata/>)
-      _ = props.save((bagParentDir / "deposit.properties").toJava) // N.B. the first write action
-      _ = if (fromVault) addAmdXml(datasetId, metadata)
-      amdChanges <- configuration.userTransformer.transform(metadata / "amd.xml")
-      _ = provenance.collectChangesInXmls(Map(
-        "http://easy.dans.knaw.nl/easy/dataset-administrative-metadata/" -> amdChanges,
-        "http://easy.dans.knaw.nl/schemas/md/ddm/" -> compare(oldDcmi, newDcmi),
-      )).foreach(xml => (metadata / "provenance.xml").writeText(xml.serialize))
-      _ = copyMigrationFiles(metadata, migration, fromVault)
+      amdFile = metadata / "amd.xml"
```
Comment (lindareijnhoudt) [inline src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala:113]: the amd.xml has already been put here in case of VAULT bags?

Comment (jo-pol) [reply src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala:114]: The next line tries to read the amd from the bag, if not found it reads from fedora. The `userTransformer` _may_ change it, once all changes are collected in memory, the amd is (re)written anyway.

File: src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala (hunk 1 above)
Comment (jo-pol) [reply src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala]: indeed

**Pull request merged**